OpenRA.Mods.Common/Activities/Move/MoveAdjacentTo.cs

```diff
@@ -94,6 +94,10 @@ public override bool Tick(Actor self)
 				Cancel(self, true);
 			else if (!IsCanceling && targetIsValid && ShouldRepath(self, oldTargetLocation))
 			{
+				// To avoid unit continuing to follow the unattackable target around until it comes into range
+				if (self.CurrentActivity is Attack a && !a.HaveArmamentsFor(target))
+					return false;
```

Comment:
attack activity should not invade MoveAdjacent to

Reply:
tbf, looking at the code I dunno how it would be best to solve this. In general, when parent activity needs to control the cancelling of chile activities, we flag the parent activity as `ChildHasPriority = false` and add control in the parents Tick method

Reply:
If `Attack` cannot be directly called within `MoveAdjacentTo`, it seems there’s no better way to determine when to return to the parent Activity to complete further logic.
But I can keep only `self.CurrentActivity is Attack` in `MoveAdjacentTo`, and return to `Attack` to complete the check for HaveArmamentsFor. Your opinion and more feedback will be helpful.

Reply:
Pun's idea sounds correct, the parent can tick it's child activity "manually" from it's own tick method and inside there do the check.